File: solutions/beecrowd/2061/2061.cpp
#include <cstdint>
#include <iostream>
#include <string>

int main() {
    std::int16_t n, m, open_tabs;
    std::string command;

    while (std::cin >> n >> m) {
        open_tabs = n;

        for (std::int16_t i = 0; i < m; i++) {
            std::cin >> command;

            if (command == "fechou") {
                open_tabs++;
            } else {
                open_tabs--;
            }

            if (open_tabs < 0) {
                open_tabs = 0;
            }
        }

        std::cout << open_tabs << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/2061/2061.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

while line := input().strip():
    n, m = map(int, line.split())

    open_tabs = n

    for _ in range(m):
        command = input().strip()

        if command == 'fechou':
            open_tabs += 1
        else:
            open_tabs -= 1

        if open_tabs < 0:
            open_tabs = 0

    output(f'{open_tabs}\n')
